### ISSUE TITLE
Very slow git repository downloads

### ISSUE THREAD

**Issue by wiktortomczak** [bazelbuild/bazel#2696]
Bazel repeatedly hangs when fetching large external git repositories (defined in the WORKSPACE file with git_repository). At some point the object count reported in the console (as in INFO: Cloning https://github.com/grpc/grpc.git: Receiving objects (171939 / 255634)) gets stuck and does not increase for 10+ minutes, or even for hours. This problem occurs non-deterministically, however when it does, it typically does at the same object count (eg. 171939).

I think this occurs more often when the network is slow or when the git hosting service (github) is under heavy load (?). Also, the cause might be in git: Cloning the same repository directly with git command-line client (git clone https://github.com/grpc/grpc.git) also hangs randomly.

Possibly related issues:
https://github.com/bazelbuild/bazel/issues/1760
https://groups.google.com/forum/#!msg/bazel-discuss/5LOE_kYkoyo/rkOLjMVaCAAJ. 

As a workaround, would it be possible to have Bazel support shallow git clones, without the history of older commints? See git clone --depth=0 and git archive.

**Comment by meteorcloudy** [comment]
//cc @kchodorow Can you take a look?

**Comment by kchodorow** [comment]
Yes, right now we're using jGit for the git_repository backend, which is suboptimal.  If you're cloning from GitHub, I highly recommend using http_archive() to download the tarball instead: it has retries, timeouts, and will essentially give you a "shallow clone."

**Comment by wiktortomczak** [comment]
Thanks, I'll try http_archive().

**Comment by wiktortomczak** [comment]
http_archive() works much better indeed.
